servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -186,6 +198,80 @@ public final HttpServerBuilder appendConnectionAcceptorFilter(final ConnectionAc
+     * <pre>
+     *     filter2 ⇒ filter1 ⇒ filter3 ⇒ service
+     * </pre>
+     *
+     * @param <FF> Super-type of the filter factory which must implement both {@link StreamingHttpServiceFilterFactory}
+     * and {@link HttpExecutionStrategyInfluencer} though non-offloading filters must not influence strategy.
+     * @param factory {@link StreamingHttpServiceFilterFactory} to append.
+     * @return {@code this}
+     * @throws IllegalArgumentException if the provided filter requires offloading.
+     */
+    public final <FF extends StreamingHttpServiceFilterFactory & HttpExecutionStrategyInfluencer> HttpServerBuilder
+                appendNonOffloadingServiceFilter(final FF factory) {
+        requireNonNull(factory);
+        HttpExecutionStrategy factoryStrategy = influenceStrategy(factory, OFFLOAD_NONE_STRATEGY);
+        if (!OFFLOAD_NONE_STRATEGY.equals(factoryStrategy)) {
+            throw new IllegalArgumentException("Non-offloading filter required offloading " + factoryStrategy);
+        }
+        noOffloadServiceFilters.add(factory);
+        return this;
+    }
+
+    /**
+     * Appends a non-offloading filter to the chain of filters used to decorate the {@link StreamingHttpService} used
+     * by this builder, for every request that passes the provided {@link Predicate}. Filters added via this method
+     * will be executed before offloading occurs and before filters appended via
+     * {@link #appendServiceFilter(StreamingHttpServiceFilterFactory)}.
+     * <p>
+     * Note this method will be used to decorate the {@link StreamingHttpService} passed to
+     * {@link #listenStreaming(StreamingHttpService)} before it is used by the server.
+     * The order of execution of these filters are in order of append, before the filters added with
+     * {@link #appendServiceFilter(StreamingHttpServiceFilterFactory)}. If 3 filters are added as follows:
+     * <pre>
+     *     builder.appendServiceFilter(filter1).appendNonOffloadingServiceFilter(filter2).appendServiceFilter(filter3)
+     * </pre>
+     * accepting a request by a service wrapped by this filter chain, the order of invocation of these filters will be:
+     * <pre>
+     *     filter2 ⇒ filter1 ⇒ filter3 ⇒ service
+     * </pre>
+     *
+     * @param <FF> Super-type of the filter factory which must implement both {@link StreamingHttpServiceFilterFactory}
+     * and {@link HttpExecutionStrategyInfluencer} though non-offloading filters must not influence strategy.
+     * @param predicate the {@link Predicate} to test if the filter must be applied. This must not block.
+     * @param factory {@link StreamingHttpServiceFilterFactory} to append.
+     * @return {@code this}
+     * @throws IllegalArgumentException if the provided filter or predicate requires offloading.
+     */
+    public final <FF extends StreamingHttpServiceFilterFactory & HttpExecutionStrategyInfluencer> HttpServerBuilder
```

Comment:
`FF` isn't used. should we avoid the generics and just address the `HttpExecutionStrategyInfluencer` being explicit in a followup PR to remote consistent with `appendServiceFilter` (question for both `appendNonOffloadingServiceFilter` methods) ?

Reply:
Removed for now.